src/main/java/com/sinthoras/visualprospecting/database/ClientCache.java

```diff
@@ -73,10 +83,18 @@ public void putUndergroundFluids(List<UndergroundFluidPosition> undergroundFluid
         for(UndergroundFluidPosition undergroundFluidPosition : undergroundFluids) {
             DimensionCache.UpdateResult updateResult = putUndergroundFluids(undergroundFluidPosition);
             if(updateResult == DimensionCache.UpdateResult.New) {
-                newUndergroundFluids++;
+            	ProspectingEvent.UndergroundFluid event = new ProspectingEvent.UndergroundFluid(undergroundFluidPosition);
```

Comment:
How do you envision that? The event can only be fired after each *Position has been checked if it is not already known, so there would be additional code required to collect a list first and fire the event afterwards.

Also, I changed the Event to non-cancelable because canceling would only killed the chat notification anyways.